src/hotspot/share/opto/loopopts.cpp

```diff
@@ -4182,6 +4182,18 @@ void PhaseIdealLoop::move_unordered_reduction_out_of_loop(IdealLoopTree* loop) {
 
     UnorderedReductionNode* last_ur = phi->in(2)->as_UnorderedReduction();
 
+    // Determine types
+    const TypeVect* vec_t = last_ur->vect_type();
+    uint vector_length    = vec_t->length();
+    BasicType bt          = vec_t->element_basic_type();
+    const Type* bt_t      = Type::get_const_basic_type(bt);
+
+    if (!last_ur->make_normal_vector_op_implemented(vec_t)) {
```

Comment:
Some naming comments,   _make_ prefix is more suitable for IR creation routines.